frontend/cs/scs/api.go

```diff
@@ -657,6 +702,22 @@ func (builder *builder) Commit(v ...frontend.Variable) (frontend.Variable, error
 	})
 }
 
+// NewCombination in the form of res = qL.a + qR.b + qM.ab + qC
+func (builder *builder) NewCombination(a, b frontend.Variable, qL, qR, qM, qC int) frontend.Variable {
```

Comment:
suggestion;
rename to `api.EvaluatePlonkExpression` + in this function, check that a, b are not constant and do the appropriate path if they are. (else this function will panic). 

also add a corresponding interface so that in a circuit, one can check if `_, ok := api.(...); ok` and have a fail over path for R1CS for example